lib/pdk/util/bundler.rb

```diff
@@ -0,0 +1,129 @@
+require 'bundler'
+require 'tty-spinner'
+require 'pdk/util'
+require 'pdk/cli/exec'
+
+module PDK
+  module Util
+    module Bundler
+      class BundleHelper; end
+
+      def self.ensure_bundle!
+        bundle = BundleHelper.new
+
+        if bundle.has_gemfile?
+          if !bundle.locked?
+            bundle.lock
+          end
+
+          if !bundle.installed?
+            bundle.install
+          end
+        end
+      end
+
+      class BundleHelper
+        def has_gemfile?
+          # return a pure boolean
+          !!gemfile
+        end
+
+        def locked?
+          # return a pure boolean
+          !!gemfile_lock
+        end
+
+        def installed?
+          output_start(_("Checking for missing Gemfile dependencies"))
+
+          result = invoke('check', "--gemfile=#{gemfile}", "--path=#{pdk_cachedir}")
```

Comment:
:+1: to having a general cache dir for the PDK. To prevent possible conflicts, I think we should standardise on `#{pdk_cachedir}/<toolname>` (in case some other tool that we want to integrate also wants to create a `ruby` directory in its cache for example).